app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,40 @@
 class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+
   def index
-    @user = User.find(params[:user_id])
+    @user = current_user
     @posts = @user.posts
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(title: params[:post][:title], text: params[:post][:text], comments_counter: 0, likes_counter: 0)
```

Comment:
- [ ] To avoid repeating ourselves let's make use of the strong parameters as defined in the `post_params` method below:

  ```
  @post = Post.new(post_params)
  ```

Reply:
@DammyShittu I have tried using post_params but when creating the post, it return errors of "Validation failed: Comments counter is not a number, Likes counter is not a number". Its because of these reasons that I m using the those params especially for comments_counter and like_counters"